Add ndvithresholding (#235)
up42/docs: source/up42-blocks/processing/up42-ndvithresholding.rst

* ``threshold_values``: A list of a dictionary with specific land type and corresponding threshold values.

Land types and values for thresholding
--------------------------------------
Below is the default threshold values that is been set for the block. However, you can change ``threshold_values`` based on your
preferences and case study.

.. table:: Default threshold values
    :align: center

   +-----------------------------+---------------------------------+
   | Land type                   | Threshold                       |
   +=============================+=================================+
   |        no_vegetation        | 0.2                             |
   +-----------------------------+---------------------------------+
   |       sparse_vegetation     | 0.4                             |
   +-----------------------------+---------------------------------+
   |      moderate_vegetation    | 0.6                             |
   +-----------------------------+---------------------------------+
   |       dense_vegetation      | 0.9                             |
   +-----------------------------+---------------------------------+

Example queries
---------------

> Example using ``n_sieve_pixels`` and ``threshold_values``:

Reviewer: where are the values for this blocks params? This seems to be about the OA basemap. No? 🤔

Author: No Antonio, it is for ndvithresholding block that I am currently working, the reason that I made this PR as a Draft was simply because it is still incomplete. I just wanted to have it here so when I finished my implementation little work remains for me.

.. meta::
   :description: UP42 processing blocks: Thresholding NDVI values
   :keywords: UP42, processing, NDVI, thresholding, vegetation, SPOT 6/7, Pléiades

.. _up42-ndvithresholding-block:

> NDVI Threshold

Reviewer: Isn't it a bit _funny_ that the page title and the filename are not aligned? Shouldn't it be: **NDVI Thresholding**. Also is an **action** not an object.

Author: Right, but now block has been published with `NDVI Threshold` title. I can change the filename here to be consistent.

> ===================================

Reviewer 2: Please adjust length. If you run make_html this likely produces a warning.

> For more information, please read the ...

Reviewer 2: Guess this needs to be updated?

Author: I think this is always your favorite part of cm on docs! yes, it will be!

Block type: ``PROCESSING``

Supported parameters
--------------------

> This block applies thresholding for NDVI values. This block is mainly suitable to be used with `Pleiades Download block <https://docs.up42.com/up42-blocks/data/pleiades-download.html>`_

Reviewer 2: This section is misleading. The input needs to be NDVI and does not depend too much on the underlying dataset used to create it.
Also: we initially discussed that this block does not only support our NDVI block, but also those implemented by partners. Was this plan cancelled?

Author: Right, I was just thinking if I only mention ndvi block before, a user would ignore the stuff that needs to be added before. Also, I mentioned the word "mainly" so I guess it is saying that it is not only working with our ndvi. I will add a sentence to be precise here.

> However, other blocks on the marketplace that provide ndvi values can also be used with this block.

Reviewer: NDVI (is an acronym)

Author: Ok, will fix it.

> * ``n_sieve_pixels``: Number of the connected pixels (small polygon) that will be replaced in output with value of their largest neighbor.

Reviewer: in the output.

Author: Good catch

* ``threshold_values``: A list of a dictionary with specific vegetation type and corresponding threshold values.

Vegetation types and values for thresholding
--------------------------------------------

> Below are the default threshold values that were been set for the block. However, you can change ``threshold_values`` based on your

Reviewer 3: [...] values that **have** been set for the block [...]

Author: Thank you :D my english is problematic here

> * ``threshold_values``: A list of a dictionary with specific land type and corresponding threshold values.

Reviewer 2: I assume this is just a dictionary and not a list of dictionaries?

Author: It is indeed a dict, but having only a dict as a parameter is not supported in specification (https://github.com/up42/specifications/blob/master/v2/blocks/parameters-schema.json), so I had to come up with a workaround and therefore I put it in a list. I can ignore mentioning here! I just want to be exact.

Reviewer 2: Exact is good, having a list of dicts is not that great, but not a topic for docu review (but for the code). Seems I missed that one, so from the docu side go ahead.

> Land types and values for thresholding

Reviewer 2: Maybe better vegetation types?

> Below is the default threshold values that is been set for the block. However, you can change ``threshold_values`` based on your

Reviewer 2: ```suggestion
Below are the default threshold values that were set for the block. However, you can change ``threshold_values`` based on your
```

.. table:: Default threshold values
    :align: center

   +-----------------------------+---------------------------------+
   | Vegetation type             | Threshold                       |
   +=============================+=================================+
   |        no_vegetation        | 0.2                             |
   +-----------------------------+---------------------------------+
   |       sparse_vegetation     | 0.4                             |
   +-----------------------------+---------------------------------+
   |      moderate_vegetation    | 0.6                             |
   +-----------------------------+---------------------------------+
   |       dense_vegetation      | 0.9                             |

>    +-----------------------------+---------------------------------+

Reviewer: You can only have **4** classes and the only degree of freedom is on the range for each?

Author: No people can add more classes if they want. The code will handle this dictionary in a generic way.

Author: I added also vegetation types to make it more clear that people can add more classes
